sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/MethodOverloadsShouldBeGroupedBase.cs

```diff
@@ -90,11 +82,12 @@ protected void CheckMembers(SyntaxNodeAnalysisContext c, IEnumerable<TMemberDecl
                 if (GetMethodName(member, IsCaseSensitive) is string methodName
                     && IsValidMemberForOverload(member))
                 {
-                    // #4136 Should not raise when static methods are grouped together
-                    key = methodName + "/" + IsStatic(member);  
+                    // Groups that should be together are defined by accessibility, abstract, static and member name #4136
+                    var accessibility = c.SemanticModel.GetDeclaredSymbol(member)?.DeclaredAccessibility.ToString();
+                    key = $"{accessibility}/{IsStatic(member)}/{IsAbstract(member)}/{methodName}";
```

Comment:
I believe it would be better now to use a dedicated utility class than a simple string as key for grouping misplaced overloads, as it is now comprised of many different attributes.

Reply:
I consider the `key` as trivial "hash" for grouping. IEqualityComparer and GetHashCode (with the same concatenation anyway) seems like a lot of extra code and state handling in order to do the same thing anyway.

What I actually don't like in this line is too much concatenation with the same separator. That could be addressed like this:
`key = new object[] { accessibility, IsStatic(member), IsAbstract(member), methodName }.JoinStr("/", x => x?.ToString());`

Reply:
I changed my mind. In bigger picture, it indeed is better to create dedicated class.